src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -544,7 +542,13 @@ private static void FreePinnedHeaders(List<GCHandle> pinnedHeaders)
 
         public void Abort()
         {
-            // TODO
+            // Use code has 
+            if (Interlocked.Exchange(ref _requestAborted, 1) == 0)
+            {
+                // call native method to abort the request
+                NativeMethods.http_abort_request(_pHttpContext);
```

Comment:
I'm assuming this also returns an HResult that needs to be checked. Kestrel does this in a very low level part of the p-invoke layer so that it's central. The check method is used around every single pinvoke https://github.com/aspnet/KestrelHttpServer/blob/204cf7a1dc9d5f9a999df3f9f288b4b535b3132a/src/Kestrel.Transport.Libuv/Internal/Networking/LibuvFunctions.cs so exceptions don't slip through the cracks. We still need to be careful where we throw to make sure we don't corrupt anything

Reply:
Yeah I think we should do this. We shouldn't throw when it is required that we return a REQUEST_NOTIFICATION_STATUS to IIS.

Reply:
https://github.com/aspnet/IISIntegration/issues/458 let's make it part of this.